src/main/scala/EnsimePlugin.scala

```diff
@@ -659,13 +672,16 @@ object SExpFormatter {
   // a lot of legacy key names and conventions
   def toSExp(m: EnsimeModule): String = s"""(
    :name ${toSExp(m.name)}
-   :source-roots ${fsToSExp((m.mainRoots ++ m.testRoots))}
-   :targets ${fsToSExp(m.targets)}
-   :test-targets ${fsToSExp(m.testTargets)}
    :depends-on-modules ${ssToSExp(m.dependsOnNames.toList.sorted)}
-   :compile-deps ${fsToSExp(m.compileJars)}
-   :runtime-deps ${fsToSExp(m.runtimeJars)}
-   :test-deps ${fsToSExp(m.testJars)}
    :doc-jars ${fsToSExp(m.docJars)}
-   :reference-source-roots ${fsToSExp(m.sourceJars)})"""
+   :reference-source-roots ${fsToSExp(m.sourceJars)}
+   :configurations ${csToSExp(m.configurations.values)})"""
+
+  def toSExp(f: EnsimeConfiguration): String = s"""
+    (:name ${f.name}
+     :source-roots ${fsToSExp(f.roots)}
+     :targets ${fsToSExp(f.targets)}
+     :compiler-args ${ssToSExp(f.compilerArgs)}
```

Comment:
it would be good to rename these things when we have the change, e.g. `scalac-options` instead of `compiler-args`